src/main/java/mods/railcraft/common/carts/CartBaseLogicChest.java

```diff
@@ -22,7 +23,7 @@
 import net.minecraft.world.IInteractionObject;
 import net.minecraft.world.World;
 
-public abstract class CartBaseLogicChest extends CartBaseLogic implements IInteractionObject {
+public abstract class CartBaseLogicChest extends CartBaseLogic implements IInteractionObject, IInventoryImplementor {
```

Comment:
I think what its actually missing is the IItemHandler capability function implementations.